src/ht.c

```diff
@@ -106,47 +92,67 @@ SDB_API bool ht_delete_internal(SdbHt* ht, const char* key, ut32* hash) {
 }
 
 SDB_API SdbHt* ht_new(DupValue valdup, HtKvFreeFunc pair_free, CalcSize calcsizeV) {
-	return internal_ht_new (ht_primes_sizes[0], (HashFunction)sdb_hash, 
-	  			(ListComparator)strcmp, (DupKey)strdup,
-				valdup, pair_free, (CalcSize)strlen, calcsizeV);
+	return internal_ht_new (ht_primes_sizes[0], 0, (HashFunction)sdb_hash,
+		(ListComparator)strcmp, (DupKey)strdup,
+		valdup, pair_free, (CalcSize)strlen, calcsizeV);
 }
 
-SDB_API void ht_free(SdbHt* ht) {
-	if (ht) {
-		ut32 i;
-		for (i = 0; i < ht->size; i++) {
-			ls_free (ht->table[i]);
-		}
-		free (ht->table);
-		ls_free (ht->deleted);
-#if INSERTORDER
-		ls_free (ht->list);
-#endif
-		free (ht);
+SDB_API SdbHt* ht_new_size(ut32 initial_size, DupValue valdup, HtKvFreeFunc pair_free, CalcSize calcsizeV) {
+	ut32 sz;
+	int i = 0;
+
+	while (i < S_ARRAY_SIZE (ht_primes_sizes) &&
```

Comment:
catch teh value of S_ARRAY_SIZE once before the while and use it in the conditional below too

Reply:
what do you mean? I don't get it.